xwiki-rendering-api/src/main/java/org/xwiki/rendering/internal/parser/reference/GenericLinkReferenceParser.java

```diff
@@ -146,14 +146,13 @@ public ResourceReference parse(String rawReference)
         }
 
         // Step 3: If we're in non wiki mode, we consider the reference to be a URL.
-        if (!isInWikiMode()) {
+        else{
             resourceReference = new ResourceReference(rawReference, ResourceType.URL);
             resourceReference.setTyped(false);
             return resourceReference;
         }
 
-        // Step 4: Consider that we have a reference to a document
-        return parseDocumentLink(content);
+        
```

Comment:
What exactly makes you think it's unnecessary ? From what I understand step 4 is actually what happen most of the time (we almost never have interwiki links and in the context of a running XWiki instance you are on wiki mode).

Seems to me you are over thinking this. You are not supposed to modify the logic of the method, just improve the codestyle.